Python/Filters/tdms_to_h5.py

```diff
@@ -0,0 +1,285 @@
+import numpy as np
+import os
+import h5py
+import nptdms
+import re
+
+from contextlib import ExitStack
+from pathlib import Path
+from typing import Any, AnyStr, Dict, Generator, List, Match, Pattern
+from enum import IntEnum
+from typing import List, Tuple, Union
+
+from dream3d.Filter import Filter, FilterDelegatePy
+from dream3d.simpl import *
+
+FILE_VERSION: int = 3
+VERSION_KEY: str = 'Version'
+SLICES_KEY: str = 'TDMSData'
+INDEX_KEY: str = 'Index'
+LAYER_START_TIME_KEY: str = 'LayerStartTime'
+LAYER_END_TIME_KEY: str = 'LayerEndTime'
+PART_START_TIME_KEY: str = 'PartStartTime'
+PART_END_TIME_KEY: str = 'PartEndTime'
+TDMS_GROUP_NAME_KEY: str = 'TDMS_GroupName'
+VERTICES_KEY: str = 'Vertices'
+  
+ def get_file_list(end_index: int, file_extension: str, file_prefix: str, file_suffix: str, increment: int, input_path: str, ordering: int, padding_digits: int, start_index: int, status_delegate: Union[FilterDelegateCpp, FilterDelegatePy] = FilterDelegatePy()) -> None:
+  file_list = []
+
+  index = 0
+  i = 0
+  while i < (end_index - start_index) + 1:
+
+    if(ordering):
+      index = end_index - i
+    else:
+      index = start_index + i
+
+    file = input_path + '/' + file_prefix + str(index).zfill(padding_digits) + file_suffix + '.' + file_extension
+    file_list.append(file)
+    i = i + increment
+
+  return file_list
+
+def tdms2h5(file_info: StackFileListInfo, file_list: List, output_dir: Path, area_offset: int, intensity_offset: int, laser_offset: int, groups: List[str] = [], status_delegate: Union[FilterDelegateCpp, FilterDelegatePy] = FilterDelegatePy()) -> None:
+
+  largest_offset = max(area_offset, intensity_offset)
+
+  with ExitStack() as exitStack:
+    h5_files: Dict[str, h5py.File] = {}
+    slice_indices: List[int] = []
+
+    regex_name: Pattern[AnyStr] = re.compile(fr'{file_info.FilePrefix}(\d+)')
+
+    path: Path
+
+    paths = []
+    for file in file_list:
+      paths.append(Path(file))
+
+    status_delegate.notifyStatusMessage(f' Files are {file_list}')

```

Comment:
If we have 1000 files (which is not out of the ordinary this is going to print a lot of information. And when it does print it will not really be easily readable. I would say maybe we should remove this line or just print a summary such as "converting XXX number of files from directory '{input_dir}' and placing into '{output_dir}'. Think that gives enough information to the user to give an idea of what is about to happen.